runtime/vm/MHInterpreter.cpp

```diff
@@ -1507,48 +1507,36 @@ VM_MHInterpreter::foldForFoldArguments(j9object_t methodHandle)
 	/* Count the slot number of all the arguments before the specified fold position to
 	 * determine the 1st slot of fold arguments.
 	 */
-	U_32 argumentSlots = getArgSlotsBeforePosition(argumentTypes, foldPosition);
+	U_32 slotsBeforeFold = getArgSlotsBeforePosition(argumentTypes, foldPosition);
 
 	/* [... foldHandle args descriptionBytes MethodTypeFrame] */
-	UDATA *spPriorToFrameBuild = _currentThread->sp;
 	(void)buildMethodTypeFrame(_currentThread, foldType);
 
-	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle args] */
-	_currentThread->sp -= (foldArgSlots  + 1);
-	memcpy(_currentThread->sp, spPriorToFrameBuild, sizeof(UDATA) * (foldArgSlots+1));
-
-	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle PlaceHolderFrame foldHandle args] */
-	insertPlaceHolderFrame(foldArgSlots, methodHandle, J9VMJAVALANGINVOKEMETHODHANDLE_FOLDHANDLEPLACEHOLDER_METHOD(_vm));
-
 	/* Get combinerHandle, combinerHandle.type and combinerHandle.type.argSlots */
 	j9object_t combinerHandle = getCombinerHandleForFold(methodHandle);
 	j9object_t combinerType = getMethodHandleMethodType(combinerHandle);
 	U_32 combinerArgSlots = getMethodTypeArgSlots(combinerType);
 
-	/* Replace foldHandle with combinerHandle and adjust sp according to combinerHandle.type.argSlots */
-	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle PlaceHolderFrame combinerHandle args] */
-	((j9object_t *)_currentThread->sp)[foldArgSlots] = combinerHandle;
-	UDATA *spCombinerSlot = _currentThread->sp + foldArgSlots;
+	/* Add combinerHandle to top of stack and adjust sp to make room for combinerHandle.type.argSlots */
+	/* [... foldHandle args descriptionBytes MethodTypeFrame combinerHandle combinerArgs] */
+	_currentThread->sp -= (combinerArgSlots  + 1);
+	((j9object_t *)_currentThread->sp)[combinerArgSlots] = combinerHandle;
```

Comment:
Is this required?  If it's overwritten by the insertPlaceHolderFrame below, then I think we can skip it here

Reply:
you're right its not required. I've removed a similar line in `filterArgumentsWithCombiner` as well